src/deep-fs/test/Local/S3FSRelativeFSExtender.js

```diff
@@ -0,0 +1,549 @@
+'use strict';
+
+import chai from 'chai';
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+import {S3FSRelativeFSExtender} from '../../lib.compiled/Local/S3FSRelativeFSExtender';
+import path from 'path';
+import nodeFS from 'fs';
+
+chai.use(sinonChai);
+
+suite('Local/S3FSRelativeFSExtender', function() {
+
+  let s3FSRelativeFSExtender = null;
+  let relativeFsPath = process.cwd() + '/test';
+  let actualResult = null;
+
+  test('Class S3FSRelativeFSExtender exists in Local/S3FSRelativeFSExtender', function() {
+    chai.expect(typeof S3FSRelativeFSExtender).to.equal('function');
+  });
+
+  test('Check _readdirp() returns valid [] of files ib dir', function() {
+    let expectedResult = [
+      'deep-framework/src/deep-fs/test/.gitkeep',
+      'deep-framework/src/deep-fs/test/FS.js',
+      'deep-framework/src/deep-fs/test/Local',
+      'deep-framework/src/deep-fs/test/Local/S3FSRelativeFSExtender.js',
+      'deep-framework/src/deep-fs/test/common',
+      'deep-framework/src/deep-fs/test/common/KernelFactory.js',
+      'deep-framework/src/deep-fs/test/common/backend-cfg-json.js',
+    ];
+    let dir = process.cwd() + '/test';
+
+    actualResult = S3FSRelativeFSExtender._readdirp(dir);
+
+    for (let i = 0; i < expectedResult.length; i++) {
+      chai.expect(actualResult[i]).to.contains(expectedResult[i]);
+    }
+  });
+
+  test('Check _readdirp() returns empty array for invalid dir', function() {
+    let dir = process.cwd() + '/test/invalid';
+    let error = null;
+
+    try {
+      S3FSRelativeFSExtender._readdirp(dir);
+    } catch (e) {
+      error = e;
+    }
+
+    chai.expect(error).to.be.an.instanceof(Error);
+  });
+
+  test('Check constructor throws exception for invalid relativeFsPath', function() {
+    s3FSRelativeFSExtender = new S3FSRelativeFSExtender(relativeFsPath);
+
+    chai.expect(s3FSRelativeFSExtender.relativeFs.constructor.name).to.eql('Object');
+    chai.expect(s3FSRelativeFSExtender.cwd).to.eql(relativeFsPath);
+  });
+
+  test('Check constructor throws exception for invalid relativeFsPath', function() {
+    let relativeFsPath = {
+      path: 'testPath/To',
+    };
+    let error = null;
+
+    try {
+      s3FSRelativeFSExtender = new S3FSRelativeFSExtender(relativeFsPath);
+    } catch (e) {
+      error = e;
+    }
+
+    chai.expect(error).to.be.an.instanceof(Error);
+  });
+
+  test(
+    'Check relativeFsExtended() copies the prototype of this.EXTEND_OBJECT to this._relativeFsObject',
+    function() {
+      chai.expect(s3FSRelativeFSExtender.relativeFsExtended.constructor.name).to.eql('Object');
+      chai.expect(typeof s3FSRelativeFSExtender.relativeFsExtended.getPath).to.eql('function');
```

Comment:
['getPath', 'clone', ...].forEach(function(method) {chai.expect(typeof s3FSRelativeFSExtender.relativeFsExtended[method]).to.eql('function');})